components/service/glean-gradle-plugin/src/main/groovy/mozilla/components/service/glean-gradle-plugin/GleanGradlePlugin.groovy

```diff
@@ -0,0 +1,287 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import org.apache.tools.ant.taskdefs.condition.Os
+import groovy.json.JsonOutput
+import org.gradle.api.GradleException
+import org.gradle.api.Plugin
+import org.gradle.api.Project
+import org.gradle.api.Task
+import org.gradle.api.artifacts.transform.ArtifactTransform
+import org.gradle.api.internal.artifacts.ArtifactAttributes
+import org.gradle.api.tasks.Exec
+import org.gradle.api.tasks.TaskProvider
+import org.gradle.platform.base.Variant
+
+/*
+ * A helper class to extract metrics.yaml files from AAR files.
+ */
+class GleanMetricsYamlTransform extends ArtifactTransform {
+    List<File> transform(File file) {
+        def f = new File(file, "metrics.yaml")
+        if (f.exists()) {
+            return [f]
+        }
+        return []
+    }
+}
+
+class GleanPlugin implements Plugin<Project> {
```

Comment:
This really looks like an almost copy-paste of the current script. Is that right?